packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -39,32 +101,120 @@ export const runAggregation = (): ThunkAction<
   RootState,
   void,
   Actions
+> => {
+  return (dispatch) => dispatch(fetchAggregationData(1));
+};
+
+export const fetchPrevPage = (): ThunkAction<
+  void,
+  RootState,
+  void,
+  Actions
+> => {
+  return (dispatch, getState) => {
+    const {
+      aggregation: { page }
+    } = getState();
+    if (page <= 1) {
+      return;
+    }
+    dispatch(fetchAggregationData(page - 1));
+  };
+};
+
+export const fetchNextPage = (): ThunkAction<
+  void,
+  RootState,
+  void,
+  Actions
+> => {
+  return (dispatch, getState) => {
+    const {
+      aggregation: { isLast, page }
+    } = getState();
+    if (isLast) {
+      return;
+    }
+    dispatch(fetchAggregationData(page + 1));
+  };
+};
+
+export const cancelAggregation = (): ThunkAction<
+  void,
+  RootState,
+  void,
+  Actions
+> => {
+  return (_dispatch, getState) => {
+    const {
+      aggregation: { abortController },
+    } = getState();
+    abortController?.abort();
+  };
+};
+
+const fetchAggregationData = (page: number): ThunkAction<
+  void,
+  RootState,
+  void,
+  Actions
 > => {
   return async (dispatch, getState) => {
     const {
       pipeline,
       namespace,
       maxTimeMS,
       collation,
-      dataService: { dataService }
+      dataService: { dataService },
+      aggregation: { limit },
     } = getState();
 
-    const stages = pipeline.map(generateStage);
-    const options: AggregateOptions = {
-      maxTimeMS: maxTimeMS || DEFAULT_MAX_TIME_MS,
-      allowDiskUse: true,
-      collation: collation || undefined,
-    };
-    if (dataService) {
+    if (!dataService) {
+      return;
+    }
+
+    const abortController = new AbortController();
+    const signal = abortController.signal;
+    abortController.signal.addEventListener('abort', cancelAggregation, { once: true });
```

Comment:
I feel like I’m missing something here – the `abortController` here is the same one that we get form the state inside of `cancelAggregation` itself (right?), so this line is saying that we should call `cancelAggregation()`, which calls `.abort()` on the AbortController, right after that has already happened? Are these different controllers/signals?